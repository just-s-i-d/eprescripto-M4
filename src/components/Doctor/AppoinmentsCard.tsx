import { useState } from "react";
import { Card, Skeleton, Avatar, Tooltip, Image } from "antd";
import { CloseCircleTwoTone, PhoneTwoTone } from "@ant-design/icons";

import noDataPng from "@assets/no-data.png";
import CardTitle from "@components/ui/CardTitle";
import PopModal from "@components/ui/PopModal";
import profilePic from "@assets/profilepic.png";
import {
  ApiResponseData,
  ApiResponseDataType,
  AppointmentDataType,
} from "@constants/types";
import { cancelAppointment } from "@utils/Doctor";
import ErrorBoundary from "@components/ErrorBoundary";
import useStatesHook from "src/hooks/useStatesHook";
import { showToast } from "@utils/common";

type AppointmentCardPropsType = {
  appointments?: ApiResponseData<AppointmentDataType>;
  loading: boolean;
  appointmentsState: ReturnType<
    typeof useStatesHook<ApiResponseData<AppointmentDataType>>
  >;
  getDataForAppointments: () => void;
};

const AppoinmentsCard = ({
  appointments,
  loading,
  appointmentsState,
  getDataForAppointments,
}: AppointmentCardPropsType) => {
  const [open, setOpen] = useState<boolean>(false);
  const [selectedPatient, setSelectedPatient] =
    useState<ApiResponseDataType<AppointmentDataType>>();
  const handlerCancleAppointment = (
    patient: ApiResponseDataType<AppointmentDataType>,
  ) => {
    setOpen(true);
    setSelectedPatient(patient);
  };
  const confirmHandler = async () => {
    if (selectedPatient) {
      cancelAppointment(selectedPatient)
        .then((res) => {
          showToast(res.message, res.type);
          getDataForAppointments();
        })
        .catch((error) => {
          showToast(error.message, error.type);
        });
    }
  };
  return (
    <>
      <Card
        className={`w-[32%] max-xxl:h-[59vh] max-h-[59vh] max-xl:max-h-[60vh] max-xxl:w-[48%] max-xl:w-[48%] max-md:w-9/12 max-sm:w-full `}
        bordered={false}
      >
        <ErrorBoundary
          refreshComponent={() => appointmentsState.setRefresh((prev) => !prev)}
          error={appointmentsState.error}
        >
          <CardTitle>Today's Appointments</CardTitle>
          {!appointmentsState.loading && !appointmentsState.data?.length ? (
            <div className="flex flex-col items-center justify-center">
              <Image src={noDataPng} preview={false} />
              <h2 className="text-lg mt-12 text-center">No Appointments</h2>
            </div>
          ) : (
            <div className="mx-auto px-4 h-[44vh] overflow-y-auto">
              <ul className="mt-1 divide-y">
                {appointments?.map((item, idx) => (
                  <li
                    key={idx}
                    className="py-4 mb-2 px-2 bg-[#F0F3F4] rounded-lg"
                  >
                    <div className="flex items-start justify-between">
                      <div className="flex gap-3">
                        {loading ? (
                          <Skeleton.Avatar active size={48} />
                        ) : (
                          <Avatar
                            size={48}
                            src={item.attributes?.profilePic || profilePic}
                          />
                        )}
                        <Skeleton
                          loading={loading}
                          active
                          avatar={false}
                          title={false}
                          paragraph={{ rows: 2, width: [100, 100] }}
                        >
                          <div>
                            <span className="block text-base text-gray-700">
                              {item.attributes?.pName}
                            </span>
                            <span className="block text-[12px] text-gray-600">
                              {item.attributes?.timeSlot}
                            </span>
                          </div>
                        </Skeleton>
                      </div>
                      {loading ? (
                        <Skeleton.Button active />
                      ) : (
                        <span
                          className={`py-1 px-2 rounded-lg ${item.attributes?.status.toLowerCase()}`}
                        >
                          {item.attributes?.status}
                        </span>
                      )}
                    </div>
                    <div className="flex justify-between">
                      <span className="block text-sm text-gray-700 font-bold">
                        <span className="mr-1 py-1 text-gray-500 font-semibold">
                          Referrer
                        </span>
                        {item.attributes.referrer || "NA"}
                      </span>
                      <div className="flex justify-end gap-4 text-lg">
                        <Tooltip title={item.attributes?.contact}>
                          <PhoneTwoTone
                            className="rotate-90 cursor-pointer"
                            href={`tel:${item.attributes?.contact}`}
                          />
                        </Tooltip>

                        {item.attributes.status === "Pending" && (
                          <CloseCircleTwoTone
                            className="cursor-pointer"
                            onClick={() => handlerCancleAppointment(item)}
                          />
                        )}
                      </div>
                      <PopModal
                        setOpen={setOpen}
                        open={open}
                        okButtonText="Sure"
                        title="Cancel Appointment"
                        footer={true}
                        confirmHandler={confirmHandler}
                      >
                        Are you sure,you want to cancel the appoinment for{" "}
                        {selectedPatient?.attributes.pName}?
                      </PopModal>
                    </div>
                  </li>
                ))}
              </ul>
            </div>
          )}
        </ErrorBoundary>
      </Card>
    </>
  );
};

export default AppoinmentsCard;
