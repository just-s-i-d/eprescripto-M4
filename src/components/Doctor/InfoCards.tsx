import { useEffect, useState } from "react";
import { Row, Col, Card, Image, Typography, Skeleton } from "antd";

import ErrorBoundary from "@components/ErrorBoundary";
import CardTitle from "@components/ui/CardTitle";
import { ApiResponseData, InfoCardDetailsType } from "@constants/types";
import { CARD_PROPERTIES } from "@constants/constants";
import useStatesHook from "../../hooks/useStatesHook";

const { Text } = Typography;

type InfoCardsProps = {
  cardDetails?: ApiResponseData<InfoCardDetailsType>;
  infoCards: ReturnType<
    typeof useStatesHook<ApiResponseData<InfoCardDetailsType>>
  >;
};

const getBgColor = (index: number) => {
  switch (index + 1) {
    case 1:
      return "bg-infoCardColors-color1";
    case 2:
      return "bg-infoCardColors-color2";
    case 3:
      return "bg-infoCardColors-color3";
    case 4:
      return "bg-infoCardColors-color4";
  }
};
const InfoCards = ({ cardDetails, infoCards }: InfoCardsProps) => {
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    cardDetails && setTimeout(() => setLoading(false), 500);
  }, [cardDetails]);
  return (
    <Card
      className="info-cards-body w-full h-[22vh] max-md:h-full flex justify-center"
      bordered={false}
    >
      <ErrorBoundary
        error={infoCards.error}
        refreshComponent={() => infoCards.setRefresh((prev) => !prev)}
      >
        <Row
          gutter={30}
          className="w-full max-md:flex max-md:flex-wrap gap-y-6 justify-center max-lg:flex-nowrap"
        >
          {cardDetails?.map((element, index) => {
            return (
              <ErrorBoundary key={element.attributes.title}>
                <Col className="basis-3/12 max-md:basis-6/12 max-sm:basis-9/12">
                  <Card
                    className={`info-card h-[17vh] py-8 px-4 max-sm:py-5 max-w-[390px] max-xl:h-[16vh] max-xl:min-w-[135px] max-sm:h-[150px] ${getBgColor(
                      index,
                    )} text-white`}
                    bordered={false}
                  >
                    <div>
                      <div className={`flex items-center gap-4 mb-2 mt-2`}>
                        {loading ? (
                          <Skeleton.Avatar active shape="square" size={50} />
                        ) : (
                          <Image
                            src={CARD_PROPERTIES[index]}
                            alt="Icon image"
                            className="w-[50px] max-xxl:w-[35px] max-md:w-[40px]"
                            preview={false}
                          />
                        )}
                        {loading ? (
                          <Skeleton.Button />
                        ) : (
                          <Text className="text-4xl max-xl:text-3xl text-white max-md:text-4xl">
                            {element.attributes.totalCount}
                          </Text>
                        )}
                      </div>
                      {loading ? (
                        <Skeleton.Input active />
                      ) : (
                        <CardTitle className="max-xxl:text-lg max-xl:text-base max-md:text-lg">
                          {element.attributes.title}
                        </CardTitle>
                      )}
                    </div>
                  </Card>
                </Col>
              </ErrorBoundary>
            );
          })}
        </Row>
      </ErrorBoundary>
    </Card>
  );
};
export default InfoCards;
