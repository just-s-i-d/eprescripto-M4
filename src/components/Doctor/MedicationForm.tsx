import { Input, Select, Form } from "antd";
import { CloseOutlined } from "@ant-design/icons";
import { MedicationType } from "@constants/types";

const { Item } = Form;
type MedicationFormPropsType = {
  fields: MedicationType;
  removeable?: boolean;
  removeMedicationInput: (id: number) => void;
};
const MedicationForm = ({
  fields,
  removeMedicationInput,
  removeable,
}: MedicationFormPropsType) => {
  return (
    <>
      <div className="mt-4 mb-2 flex justify-between">
        <h2 className="text-xl">Medication</h2>
        {removeable && (
          <CloseOutlined
            onClick={() => removeMedicationInput(fields.id)}
            data-testid="remove-field"
            className="text-[16px]"
          />
        )}
      </div>

      <div className="w-full grid sm:grid-cols-2 md:grid-cols-4 gap-x-5 gap-y-3">
        <Item
          name={fields.medicine}
          label="Medicine Name"
          className="w-full"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[a-zA-Z0-9.-]+$/,
              message: "Special characters not allowed",
            },
          ]}
        >
          <Input />
        </Item>
        <Item
          name={fields.dosage}
          label="Dosage"
          className="w-full"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[0-9/mg]/,
              message: "Enter a valid dosage",
            },
          ]}
        >
          <Input />
        </Item>
        <Item
          name={fields.timesPerDay}
          label="Frequency"
          className="w-full"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
          ]}
        >
          <Select
            options={[
              {
                value: 1,
                label: "1/Day",
              },
              {
                value: 2,
                label: "2/Day",
              },
              {
                value: 3,
                label: "3/Day",
              },
            ]}
          />
        </Item>
        <div className="flex gap-6">
          <Item
            name={fields.instruction}
            className="w-full"
            label="Instruction"
            required
            rules={[
              {
                required: true,
                message: "Field cannot be empty",
              },
            ]}
          >
            <Select
              options={[
                {
                  value: "beforeFood",
                  label: "Before eating food",
                },
                {
                  value: "afterFood",
                  label: "After eating Food",
                },
              ]}
            />
          </Item>
        </div>
      </div>
    </>
  );
};
export default MedicationForm;
