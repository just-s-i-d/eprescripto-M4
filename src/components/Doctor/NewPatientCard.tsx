import ErrorBoundary from "@components/ErrorBoundary";
import { Form, FormInstance, Input, Select } from "antd";
import FormItem from "antd/es/form/FormItem";
import TextArea from "antd/es/input/TextArea";

type NewPatientCardPropType = {
  newPatientForm: FormInstance;
};
const NewPatientCard = ({ newPatientForm }: NewPatientCardPropType) => {
  return (
    <ErrorBoundary>
      <Form form={newPatientForm} layout="vertical" className="mt-6 px-4">
        <FormItem
          name="pName"
          label="Patient Name"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[A-Za-z\s]+$/,
              message: "Special characters not allowed",
            },
          ]}
        >
          <Input />
        </FormItem>
        <FormItem
          name="email"
          label="Email"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[\w.-]+@[\w.-]+\.\w+$/,
              message: "Enter a valid mail",
            },
          ]}
        >
          <Input />
        </FormItem>
        <div className="flex gap-6">
          <FormItem
            name="dob"
            label="Date of Birth"
            required
            className="w-full"
            rules={[
              {
                required: true,
                message: "Please enter your age!",
              },
            ]}
          >
            <Input type="date" />
          </FormItem>
          <FormItem
            name="gender"
            label="Gender"
            required
            className="w-full"
            rules={[
              {
                required: true,
                message: "Please select your gender!",
              },
            ]}
          >
            <Select
              placeholder="Select your gender"
              options={[
                { value: "male", label: "Male" },
                { value: "female", label: "Female" },
                { value: "other", label: "Other" },
              ]}
            />
          </FormItem>
        </div>
        <FormItem
          name="contactNo"
          label="Mobile"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[6-9]\d{9}$/,
              message: "Enter valid no.",
            },
          ]}
        >
          <Input />
        </FormItem>
        <FormItem name="allergies" label="Mention allergies , if any">
          <TextArea placeholder="Mention your allergies here" />
        </FormItem>
      </Form>
    </ErrorBoundary>
  );
};
export default NewPatientCard;
