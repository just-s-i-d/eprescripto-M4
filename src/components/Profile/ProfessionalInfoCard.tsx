import CardTitle from "@components/ui/CardTitle";
import { Button, Card, Form, Input, Select, Space } from "antd";
import { useState } from "react";

const ProfessionalInfoCard = () => {
  const [disabled, setDisabled] = useState(true);
  return (
    <Card>
      <CardTitle>Professional Information</CardTitle>
      <Form className="flex flex-col">
        <Space>
          <Form.Item
            name="licenseNo"
            label="Licesnse Number"
            rules={[
              {
                required: true,
                message: "Please enter your speciality!",
              },
            ]}
          >
            <Input disabled />
          </Form.Item>
          <Form.Item
            name="speciality"
            label="Speciality"
            rules={[
              {
                required: true,
                message: "Please enter your speciality!",
              },
            ]}
          >
            <Input disabled={disabled} />
          </Form.Item>
          <Form.Item
            name="experience"
            label="Experience"
            rules={[
              {
                required: true,
                message: "Please enter your experiance!",
              },
            ]}
          >
            <Input type="number" disabled={disabled} />
          </Form.Item>
        </Space>
        <Space>
          <Form.Item
            name="organizationType"
            label="Organization Type"
            rules={[
              {
                required: true,
                message: "Please select the organization type!",
              },
            ]}
          >
            <Select
              className="w-[200px]"
              disabled
              options={[
                { value: "default", label: "Select an option" },
                { value: "hospital", label: "Hospital" },
                { value: "clinic", label: "Clinic" },
              ]}
            />
          </Form.Item>
          <Form.Item
            name="organizationName"
            label="Organization Name"
            rules={[
              {
                required: true,
                message: "Please enter the Organization name!",
              },
            ]}
          >
            <Input disabled={disabled} />
          </Form.Item>
        </Space>
        <Space>
          {disabled ? (
            <Button onClick={() => setDisabled(false)}>Edit Details</Button>
          ) : (
            <>
              <Button>Save</Button>
              <Button onClick={() => setDisabled(true)}>Cancel</Button>
            </>
          )}
        </Space>
      </Form>
    </Card>
  );
};
export default ProfessionalInfoCard;
