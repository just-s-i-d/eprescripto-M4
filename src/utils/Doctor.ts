import axios from "axios";

import { getLabels } from "@constants/constants";
import {
  AppointmentDataType,
  AppointmentsDataType,
  ChartApiDataType,
  DoctorAppointmentsDataType,
  InfoCardDetailsType,
  LineChartApiResDataType,
  LineChartDataType,
  PrescrtionsDataType,
  ReviewsDataType,
} from "@constants/types";

//endpoints for data
export const infoCardDataEndPoint =
  "https://api.npoint.io/bd6151d98738374a2236";

export const appointmentEndPoint = "https://api.npoint.io/9cc072a9f2c41c77a40a";

export const prescriptionsEndPoint =
  "https://api.npoint.io/7c31657f364397ae7175";

export const pieChartDataEndPoint =
  "https://api.npoint.io/f8eeff1048038038110e";

export const lineChartDataEndPoint =
  "https://api.npoint.io/75d76a4af2cc563e20a8";

export const barChartDataEndPoint =
  "https://api.npoint.io/bb362ddf298c94c14975";

export const doctorAppointmentsEndPoint =
  "https://api.npoint.io/a7f8fadf08c1b182fc98";

export const doctorReviewsEndPoint =
  "https://api.npoint.io/d21b0ee032131e268c76";

export function formatDateReadable(dateString: string) {
  const options = { year: "numeric", month: "long", day: "numeric" };
  const formattedDate = new Date(dateString).toLocaleDateString(
    undefined,
    options,
  );
  return formattedDate;
}
export const getInfoCardsData = async (): Promise<InfoCardDetailsType> => {
  return new Promise((resolve, reject) => {
    try {
      axios
        .get(infoCardDataEndPoint)
        .then((res) => {
          resolve(res.data.infoCardsData);
        })
        .catch(() => {
          reject("Cannot get data");
        });
    } catch {
      reject("Server error");
    }
  });
};

export const getAppointmentData = async (): Promise<AppointmentsDataType> => {
  return new Promise((resolve, reject) => {
    try {
      axios
        .get(appointmentEndPoint)
        .then((res) => {
          resolve(res.data.appointments);
        })
        .catch(() => {
          reject("Cannot get data");
        });
    } catch {
      reject("Server error");
    }
  });
};

export const getChartData = async (
  endPoint: string,
): Promise<ChartApiDataType> => {
  return new Promise((resolve, reject) => {
    try {
      axios
        .get(endPoint)
        .then((res) => {
          resolve(res.data);
        })
        .catch(() => {
          reject("Cannot get data");
        });
    } catch (error) {
      reject("Server Connection Error");
    }
  });
};

export const getPrescriptions = async (): Promise<PrescrtionsDataType> => {
  return new Promise((resolve, reject) => {
    try {
      axios
        .get(prescriptionsEndPoint)
        .then((res) => {
          resolve(res.data.prescriptions);
        })
        .catch(() => {
          reject("Cannot get data");
        });
    } catch (error) {
      reject("Server Connection Error");
    }
  });
};

export const getDataForLineGraph = (
  data: LineChartApiResDataType,
  days: number = 7,
): LineChartDataType => {
  const labels = getLabels(days);
  switch (days) {
    case 7:
      return { labels, data: data.seven };
    case 10:
      return {
        labels,
        data: data.ten,
      };
    case 30:
      return {
        labels,
        data: data.thirty,
      };
    default:
      return { labels, data: data.seven };
  }
};

export const cancelAppointment = async (appointment: AppointmentDataType) => {
  try {
    const { pId } = appointment;
    const res = await axios.get(appointmentEndPoint);
    if (res.status >= 200 && res.status <= 299) {
      const appointments = res.data.appointments;
      appointments.filter(
        (patient: AppointmentDataType) => patient.pId !== pId,
      );
      return { message: "Appointment Cancelled", type: "success" };
    } else {
      return { message: "Server error", type: "error" };
    }
  } catch (error) {
    return { message: "Server Connection Error", type: "error" };
  }
};

export const getDoctorAppointmentsData =
  async (): Promise<DoctorAppointmentsDataType> => {
    return new Promise((resolve, reject) => {
      try {
        axios
          .get(doctorAppointmentsEndPoint)
          .then((res) => {
            resolve(res.data.appointments);
          })
          .catch(() => {
            reject("Cannot get data");
          });
      } catch {
        reject("Server error");
      }
    });
  };

export const getReviewsData = async (): Promise<ReviewsDataType> => {
  return new Promise((resolve, reject) => {
    try {
      axios
        .get(doctorReviewsEndPoint)
        .then((res) => {
          resolve(res.data.reviews);
        })
        .catch(() => {
          reject("Cannot get data");
        });
    } catch {
      reject("Server error");
    }
  });
};
