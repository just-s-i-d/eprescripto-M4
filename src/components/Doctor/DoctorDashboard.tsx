import { useEffect, useState } from "react";

import NextPatientCard from "./NextPatientCard";
import InfoCards from "./InfoCards";
import ErrorBoundary from "@components/ErrorBoundary";
import PieChart from "../Charts/PieChart";
import LineChart from "@components/Charts/LineChart";
import BarChart from "@components/Charts/BarChart";
import AppoinmentsCard from "./AppoinmentsCard";
import {
  AppointmentsDataType,
  ChartDataType,
  InfoCardDetailsType,
  LineChartApiResDataType,
} from "@constants/types";
import {
  barChartDataEndPoint,
  getAppointmentData,
  getChartData,
  getDataForLineGraph,
  getInfoCardsData,
  lineChartDataEndPoint,
  pieChartDataEndPoint,
} from "@utils/Doctor";

const DoctorDashboard: React.FC = () => {
  const [lineChartData, setLineChartData] = useState<LineChartApiResDataType>();
  const [infoCardError, setInfoCardError] = useState(false);
  const [appointmentCardError, setAppointmentCardError] = useState(false);
  const [appointments, setAppointments] = useState<AppointmentsDataType>();
  const [infoCardsData, setInfoCardsData] = useState<InfoCardDetailsType>();
  const [pieChartData, setPieChartData] = useState<ChartDataType>();
  const [barChartData, setBarChartData] = useState<ChartDataType>();

  useEffect(() => {
    getInfoCardsData()
      .then((res) => {
        setInfoCardsData(res);
        setInfoCardError(false);
      })
      .catch(() => setInfoCardError(true));
    getAppointmentData()
      .then((res) => setAppointments(res))
      .catch(() => setAppointmentCardError(true));
    getChartData(pieChartDataEndPoint).then((res) => {
      setPieChartData(res.pieChartData);
    });
    getChartData(lineChartDataEndPoint).then((res) => {
      setLineChartData(res.lineChartData);
    });
    getChartData(barChartDataEndPoint).then((res) => {
      setBarChartData(res.barChartData);
    });
  }, []);
  return (
    <>
      <ErrorBoundary>
        <div className="max-w-full flex">
          <div className="w-full flex flex-col gap-6">
            <ErrorBoundary error={infoCardError}>
              <InfoCards cardDetails={infoCardsData} error={infoCardError} />
            </ErrorBoundary>
            <div className="flex flex-wrap justify-between gap-y-6 max-tablet:justify-center">
              <ErrorBoundary>
                <NextPatientCard
                  appointment={appointments && appointments[0]}
                  error={appointmentCardError}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <PieChart
                  chartData={pieChartData}
                  title="Ratings Composition"
                  label="Reviews"
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <AppoinmentsCard
                  appointments={appointments}
                  loading={appointmentCardError}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <LineChart
                  chartData={lineChartData}
                  title="Patients in Last 7 Days"
                  xAxesTitle="Dates"
                  yAxesTitle="Number of Patients"
                  getDataForLineGraph={getDataForLineGraph}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <BarChart
                  chartData={barChartData}
                  title="Age Groups Comparison"
                  label="Patients"
                  xAxesTitle="Different Age Groups (In Years)"
                  yAxesTitle="Number of Patients"
                />
              </ErrorBoundary>
            </div>
          </div>
        </div>
      </ErrorBoundary>
    </>
  );
};

export default DoctorDashboard;
