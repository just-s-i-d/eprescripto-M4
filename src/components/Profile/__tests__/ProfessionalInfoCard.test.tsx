import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@test/matchMedia";
import ProfessionalInfoCard from "../ProfessionalInfoCard";
import { updateUserData } from "@utils/Doctor";

jest.mock("@utils/Doctor", () => ({
  updateUserData: jest.fn().mockResolvedValue("updated"),
}));
const mockUserData = {
  licenseNo: 1234,
  speciality: "Cardiology",
  experience: 5,
  organizationType: "hospital",
  organizationName: "ABC Hospital",
};

describe("ProfessionalInfoCard", () => {
  it("Render the card with given user data", () => {
    const { asFragment } = render(
      <ProfessionalInfoCard userData={mockUserData} />,
    );
    expect(screen.getByText("Professional Information")).toBeInTheDocument();
    expect(screen.getByLabelText("License Number")).toHaveValue(
      mockUserData.licenseNo,
    );
    expect(screen.getByLabelText("Speciality")).toHaveValue(
      mockUserData.speciality,
    );
    expect(screen.getByLabelText("Experience")).toHaveValue(
      mockUserData.experience,
    );
    expect(screen.getByLabelText("Organization Name")).toHaveValue(
      mockUserData.organizationName,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders ProfessionalInfoCard component with user data", async () => {
    render(<ProfessionalInfoCard userData={mockUserData} />);
  });

  it("allows editing and saving professional details", async () => {
    render(<ProfessionalInfoCard userData={mockUserData} />);
    fireEvent.click(screen.getByText("Edit Details"));
    fireEvent.change(screen.getByLabelText("License Number"), {
      target: { value: 5678 },
    });
    fireEvent.click(screen.getByText("Save"));
    expect(updateUserData).toHaveBeenCalledWith({
      ...mockUserData,
      licenseNo: "5678",
    });
  });

  it("cancels the editing of professional details", async () => {
    render(<ProfessionalInfoCard userData={mockUserData} />);
    fireEvent.click(screen.getByText("Edit Details"));
    fireEvent.change(screen.getByLabelText("License Number"), {
      target: { value: 5678 },
    });
    fireEvent.click(screen.getByText("Cancel"));
    expect(screen.getByText("Edit Details")).toBeInTheDocument();
    expect(screen.getByLabelText("License Number")).toHaveValue(
      mockUserData.licenseNo,
    );
  });
});
