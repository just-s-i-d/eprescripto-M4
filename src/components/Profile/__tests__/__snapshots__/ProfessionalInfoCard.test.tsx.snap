// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfessionalInfoCard Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-1adbn6x"
  >
    <div
      class="ant-card-body"
    >
      <h1
        class="text-xl font-bold mb-6 undefined"
      >
        Professional Information
      </h1>
      <form
        class="ant-form ant-form-vertical css-dev-only-do-not-override-1adbn6x custom-form custom-form-profile w-full flex flex-col gap-2 mb-4"
      >
        <div
          class="flex xs:flex-col sm:flex-row xs:gap-2 sm:gap-6 sm:w-full md:w-9/12 xl:w-8/12"
        >
          <div
            class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
              >
                <label
                  class="ant-form-item-required"
                  for="licenseNo"
                  title="License Number"
                >
                  License Number
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      aria-required="true"
                      class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                      disabled=""
                      id="licenseNo"
                      type="number"
                      value="1234"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
              >
                <label
                  class="ant-form-item-required"
                  for="speciality"
                  title="Speciality"
                >
                  Speciality
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      aria-required="true"
                      class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                      disabled=""
                      id="speciality"
                      type="text"
                      value="Cardiology"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex xs:flex-col sm:flex-row xs:gap-2 sm:gap-6 sm:w-full md:w-9/12 xl:w-8/12"
        >
          <div
            class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
              >
                <label
                  class="ant-form-item-required"
                  for="experience"
                  title="Experience"
                >
                  Experience
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      aria-required="true"
                      class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                      disabled=""
                      id="experience"
                      type="number"
                      value="5"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
              >
                <label
                  class="ant-form-item-required"
                  for="organizationType"
                  title="Organization Type"
                >
                  Organization Type
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <div
                      aria-required="true"
                      class="ant-select ant-select-in-form-item css-dev-only-do-not-override-1adbn6x ant-select-single ant-select-show-arrow ant-select-disabled"
                    >
                      <div
                        class="ant-select-selector"
                      >
                        <span
                          class="ant-select-selection-search"
                        >
                          <input
                            aria-autocomplete="list"
                            aria-controls="organizationType_list"
                            aria-expanded="false"
                            aria-haspopup="listbox"
                            aria-owns="organizationType_list"
                            aria-required="true"
                            autocomplete="off"
                            class="ant-select-selection-search-input"
                            disabled=""
                            id="organizationType"
                            readonly=""
                            role="combobox"
                            style="opacity: 0;"
                            type="search"
                            unselectable="on"
                            value=""
                          />
                        </span>
                        <span
                          class="ant-select-selection-item"
                          title="Hospital"
                        >
                          Hospital
                        </span>
                      </div>
                      <span
                        aria-hidden="true"
                        class="ant-select-arrow"
                        style="user-select: none;"
                        unselectable="on"
                      >
                        <span
                          aria-label="down"
                          class="anticon anticon-down ant-select-suffix"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="down"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-form-item xs:full sm:w-[49%] md:w-[36%] xl:w-[28%] css-dev-only-do-not-override-1adbn6x"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
            >
              <label
                class="ant-form-item-required"
                for="organizationName"
                title="Organization Name"
              >
                Organization Name
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    aria-required="true"
                    class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                    disabled=""
                    id="organizationName"
                    type="text"
                    value="ABC Hospital"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
      <div
        class="ant-space css-dev-only-do-not-override-1adbn6x ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
      >
        <div
          class="ant-space-item"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-1adbn6x ant-btn-default"
            type="button"
          >
            <span>
              Edit Details
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
